docs/first_semantic_game/02.simple_chat_npcs.rst

```diff
@@ -1,163 +1,395 @@
 Simple Chat NPCs
-================
-    
-This tutorials aims to guide you through the process of creating semantic non-player characters (NPCs) that can hold interactive conversations in your Godot engine game.
+=================
 
-Creating a Basic NPC
------------------------
+This section aims to guide you through the process of creating NPCs that 
+can hold interactive conversations.  It assumes that you have completed the 
+project setup and have configured the OPENAI_API_KEY user secret.
 
-With the Semantic Kernel set up and Godot Project running, the next step is to create a basic NPC for the player to interact with. Here we will use simple shapes for NPCs instead of more lifelike characters.
+The NPC Sprite
+--------------
 
-Step 1: Creating the NPC Sprite
-**********************************
+Here we will create a basic sprite that we can attach chat functionality to.
 
-- Switch to the 2D workspace in Godot.
-- In the Scene tab, click the "+" symbol to add a 'Node'. In the "Create New Node" window, Double click on `Area2D` to add it to the scene. This will serve as the basis of your NPC.
-- Rename the `Area2D` node to 'NPC' or any name you prefer.
-- With the 'NPC' node selected, add another new node under it. This time, choose a `CollisionShape2D`. This is important as it dictates the clickable interactable area for your NPC.
-- To create a visible entity for the NPC, create another child node under the `CollisionShape2D` called `Sprite`. Here you can upload or select any simple 2D shape image you have available.
+1. In the ``MainGame`` scene that we created in :doc:`01.project_setup`
+   insert an ``Area2D`` node and rename it to  ``NPC``.
+2. Add a ``CollisionShape2D`` rectangle to the NPC. This will be useful 
+   later and removes the yellow warning in the workspace.

```

Comment:
I think it would be fine to make this a `Node2D` at this point and have the `ChatEntity` class inherit from `Node2D` for now. We don't need the functionality of the `Area2D` right now, so not worrying about that would make this section a little simpler and clearer. It should be easy enough to change the `ChatEntity` script later so that it inherits from `Area2D` when we need that functionality.

Reply:
I just saw [<https://github.com/RaiderSoft/semantic-godot/issues/5#issuecomment-1710460506>](url) about using the NPC scene as a starting point for the Gralk scene. I think this is an option, but it might add some complexity to this first demo part.

If we did this, we'd want to make the root `NPC` node a `CharacterBody2D`, which is what it is for the Gralk scene later on. Also, we might want to attach `ChatAI` as a child node of the `NPC` node (not as a script for the `NPC` node), since that is how we do it later on. To minimize complexity, the `ChatEntity`/`ChatAI` nodes could still just inherit from `Node2D` for this section.

Reply:
Changed to Node2D